lib/crowbar/init/application.rb

```diff
@@ -382,13 +388,17 @@ class Application < Sinatra::Base
 
           if init[:body] # nil body means success
             logger.error(init[:body])
-            res[:error] = init[:body]
+            res[:errors] = {
+              data: init[:body]
+            }
             http_code = 422
           end
         end
 
-        if result[:error]
-          upgrade_status.end_step(false, database: result[:error])
+        if result[:errors] || http_code != 200
+          # reformat the error response to match common format
+          result[:errors] = { database_new: result[:errors] }
+          upgrade_status.end_step(false, result[:errors])
```

Comment:
same here: if there is a chance that result[:errors] is actually nil, then why still use it?